API/Controllers/FileController.cs

```diff
@@ -0,0 +1,222 @@
+* it under the terms of the GNU Lesser General Public License as published 
+* by the Free Software Foundation version 3 of the License.
+* 
+* This program is distributed in the hope that it will be useful, 
+* but WITHOUT ANY WARRANTY; without even the implied warranty 
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+* See the GNU Lesser General Public License for more details.
+* 
+* You can find a copy of the GNU Lesser General Public License 
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using API.Extensions;
+using API.Resources;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Models;
+using Models.Defaults;
+using Models.Exceptions;
+using Services.Services;
+using System;
+using System.IO;
+using System.Net;
+using File = Models.File;
+
+namespace API.Controllers
+{
+    /// <summary>
+    /// This class is responsible for handling HTTP requests that are related
+    /// to file uploading, for example creating, retrieving or deleting.
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FileController : ControllerBase
+    {
+
+        private readonly IFileService fileService;
+        private readonly IMapper mapper;
+        private readonly IUserService userService;
+        private readonly IFileUploader fileUploader;
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileController"/> class.
+        /// </summary>
+        /// <param name="fileService">The file service.</param>
+        /// <param name="mapper">The mapper.</param>
+        /// <param name="userService">The User service</param>
+        /// /// <param name="fileUploader">The file uploader extension</param>
+        public FileController(IFileService fileService, IMapper mapper, IFileUploader fileUploader, IUserService userService)
+        {
+            this.fileService = fileService;
+            this.mapper = mapper;
+            this.userService = userService;
+            this.fileUploader = fileUploader;
+        }
+
+        /// <summary>
+        /// Get all files
+        /// </summary>
+        /// <returns>A response and list of files.</returns>
+        /// <response code="200">This endpoint returns all projects.</response>
+        [HttpGet]
+        [Authorize]
+        [ProducesResponseType(typeof(IEnumerable<FileResourceResult>), (int) HttpStatusCode.OK)]
+        public async Task<IActionResult> GetFilesAsync()
+        {
+            IEnumerable<File> files = await fileService.GetAll();
+
+            return Ok(mapper.Map<IEnumerable<File>, IEnumerable<FileResourceResult>>(files));
+        }
+
+        /// <summary>
+        /// Uploads a single file
+        /// </summary>
+        /// <returns>HTTP Response</returns>
```

Comment:
In the returns tag you should show to return type in a success scenario. So in this example:
`This method returns status code 200` or something like this.